tests/test_quickbooks_interrupted_sync.py

```diff
@@ -0,0 +1,216 @@
+            self.assertIsNone(currently_syncing)
+
+            # Verify bookmarks are saved
+            self.assertIsNotNone(final_state.get('bookmarks'))
+
+            # Verify final_state is equal to uninterrupted sync's state
+            # (This is what the value would have been without an interruption and proves resuming succeeds)
+            self.assertDictEqual(final_state, full_sync_state)
+
+        # Stream level assertions
+        for stream in expected_streams:
+            with self.subTest(stream=stream):
+
+                # Gather expectations
+                expected_replication_method = self.expected_replication_method()[stream]
+                expected_replication_key = next(iter(self.expected_replication_keys()[stream]))
+                
+                # Gather actual results
+                full_records = [message['data'] for message in synced_records_full_sync.get(stream, {}).get('messages', [])]
+                full_record_count = record_count_by_stream_full_sync.get(stream, 0)
+                interrupted_records = [message['data'] for message in synced_records_interrupted_sync.get(stream, {}).get('messages', [])]
+                interrupted_record_count = record_count_by_stream_interrupted_sync.get(stream, 0)
+
+                # Final bookmark after interrupted sync
+                final_stream_bookmark = final_state['bookmarks'][stream]['LastUpdatedTime']
+
+                if expected_replication_method == self.INCREMENTAL:
+
+                    # Verify final bookmark saved the match formatting standards for resuming sync
+                    self.assertIsNotNone(final_stream_bookmark)
+                    self.assertIsInstance(final_stream_bookmark, str)
+                    self.assertIsDateFormat(final_stream_bookmark, "%Y-%m-%dT%H:%M:%S%z") # Bookmark is being saved with timezone
+
+                    if stream == state['currently_syncing']:
+
+                        # Check if the interrupted stream has a bookmark written for it
+                        if state["bookmarks"].get(stream):
+                            interrupted_stream_bookmark = self.convert_state_to_utc(state['bookmarks'].get(stream).get('LastUpdatedTime'))
+                            interrupted_bookmark_datetime = dt.strptime(interrupted_stream_bookmark, "%Y-%m-%dT%H:%M:%SZ")
+                        else:
+                            # Assign the start date to the interrupted stream
+                            interrupted_bookmark_datetime = start_date_datetime
+
+                        # - Verify resuming sync only replicates records with replication key values greater or
+                        #       equal to the state for streams that were replicated during the interrupted sync.
+                        # - Verify the interrupted sync replicates the expected record set all interrupted records are in full records
+                        for record in interrupted_records:
```

Comment:
Can you condense the assertions in this test? There is a lot of unnecessary conditional logic and repeated assertions. It decreases readability.

Reply:
We condensed some assertions for non-interrupted streams. But for the interrupted streams, we verify that we synced all the records >= start date and verify the count of the records based on the 1st sync.